ingest-asset-opex-creator/src/main/scala/uk/gov/nationalarchives/ingestassetopexcreator/Lambda.scala

```diff
@@ -38,6 +38,7 @@ class Lambda extends LambdaRunner[Input, Unit, Config, Dependencies] {
       _ <- log(s"Asset ${asset.id} retrieved from Dynamo")
 
       children <- childrenOfAsset(dynamoClient, asset, config.dynamoTableName, config.dynamoGsiName)
+      _ <- IO.raiseWhen(children.length != asset.childCount)(new Exception(s"Asset ${asset.id}: Expected ${asset.childCount} children but ${children.length} children found"))
```

Comment:
Even though what this error means is obvious, I was thinking about future us/employees reading it:

Do you think we should add add where the counts came from? e.g.
s"Asset ${asset.id}: Expected ${asset.childCount} children but found ${children.length} children in the Preservation system"
Or

s"Asset ${asset.id}: has ${asset.childCount} children in the files table but found ${children.length} children in the Preservation system"?

Maybe "Asset id ${asset.id}" or "'assetId' ${asset.id}" instead of "Asset ${asset.id}" just because the id and name are both uuids

Reply:
Good point, I've gone with `s"Asset id ${asset.id}: has ${asset.childCount} children in the files table but found ${children.length} children in the Preservation system"`